src/Data/YAML/Dumper.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE Safe              #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+
+module Data.YAML.Dumper
+    ( encodeNode
+    -- , encodeNode'
+    , dumpEvents
+    ) where
+
+import           Data.YAML.Event.Internal as YE
+import           Data.YAML.Internal       as YI
+import           Data.YAML.Schema         as YS
+import           Data.YAML.Event.Writer   (writeEvents)
+
+import qualified Data.ByteString.Lazy     as BS.L
+import qualified Data.Map                 as Map
+import qualified Data.Text                as T
+
+
+type EvList = [Either String Event]
+type Node2EvList = [Node ()] -> EvList
+
+-- | Dump YAML Nodes as a lazy 'BS.L.ByteString'
+--
+-- Each YAML 'Node' is emitted as a individual YAML Document where each Document is terminated by a 'DocumentEnd' indicator.
+--
+-- This is a convenience wrapper over `encodeNode'`
+--
+-- @since 0.2.0
+encodeNode :: [Doc (Node ())] -> BS.L.ByteString
+encodeNode nodes = writeEvents UTF8 $ case sequence (dumpEvents (map getDoc nodes)) of
```

Comment:
I'm not happy with this `case sequence ... of` but we'll deal with this in a different PR; I'm going to merge this PR now

Reply:
Resolved, see PR #21